zinit-autoload.zsh

```diff
@@ -2702,6 +2702,12 @@ ZINIT[EXTENDED_GLOB]=""
 # FUNCTION: .zinit-cd [[[
 # Jumps to plugin's directory (in Zinit's home directory).
 #
+# Remembers the visited dirs in a backlog and allows to automatically cd
+# to them when no argument is given ($1, $2 are empty) or if a negative
+# number will be given to point to that number distance entry from the top
+# of the stack. -1 (the default) points to the last, most recent dir, and
+# so on. Remembers ~1500 last directories.
+#
```

Comment:
```suggestion
# Remembers which directories you have navigated to, so you can "jump" to them in
# just a few keystrokes. When n ($1, $2 are empty) or if a negative number will
# be given to point to that number distance entry from the top of the stack. -1
# (the default) points to the last, most recent dir, and so on. By default, the
# last 1500 directories are remembered.
```

- Are the entries duplicated?
- What exactly happens when no arguments are passed?
- IMO, Since you can't go forward in history, it makes more sense to use positive numbers to specify which directory to `cd` to. Additionally, makes it not seem like a flag.

Reply:
does `zi cd -2 && zi cd -3` equal `zi cd -5`?

Reply:
How do you set the # of directories to be remembered? 1500 seems high as a default

Reply:
is `zi cd` equal to `cd -`?

Reply:
> does `zi cd -2 && zi cd -3` equal `zi cd -5`?  

No, the history is stable. Only CD to an explicitly given plugin changes it.

As for the positive index it would be different from `cd -?`, so I think that negative indices are better.

Reply:
1500 is hardcoded into the source. I always use high history counts so I've entered such a relatively high number (my normal history is 200k).

Yes, `zi cd` does resemble `cd -` to some extend, except that the history stack isn't suffled (i.e. after `cd -5` the 5th entry becomes the top), it's stable so that one can do `zi cd -5`, `zi cd -1`, `zi cd -5` with confidence of what will happen.

Reply:
@psprint Thanks for answering my questions.